modules/keepalived-limitation-for-ipi.adoc

```diff
@@ -0,0 +1,48 @@
+[discrete]
+[id="keepalived-limitation-for-ipi_{context}"]
+= Identifying pre-existing virtual routers
+
+With installer-provisioned infrastructure (IPI) installations, the Virtual IPs are managed using multicast (VRRPv2 or VRRPv3), and there is a limitation of 255 unique virtual routers per multicast domain. Each VIP has a virtual router id associated with it based on the cluster name. Duplication of these ids can result in a conflict, preventing the Virtual IPs from being assigned to the node. Therefore, if you have pre-existing virtual routers using the standard IPv4 or IPv6 multicast groups, you should check to see which VIPs the installation will choose to avoid conflicts.
+
+To discover the VIPs that will be chosen, you can use the runtimecfg utility.
+
+.Procedure
+
+. Run the runtimecfg utility with a domain name, in this example `cnf10`:
++
+[source, terminal]
+----
+$ podman run quay.io/openshift/origin-baremetal-runtimecfg:TAG vr-ids cnf10
```

Comment:
We need to take it from the release payload:
https://bugzilla.redhat.com/show_bug.cgi?id=1886450#c7

Reply:
> We need to take it from the release payload:
> https://bugzilla.redhat.com/show_bug.cgi?id=1886450#c7

Hi Gal,

My concern with pulling from the release payload is that the 2 commands are very version-specific (ocp-v4.0-art-dev@sha256:99f031207070be623af0cbda37e36667cbd11dae917a88479c0c6bb375dac282),
and it's better not to get stuck on a very specific version in the documentation, in case these things change more quickly than the documentation release cycle. Is there a way we can make the commands more generic? Perhaps:
oc adm release info --registry-config /home/dev-scripts/pull_secret.json registry.ci.openshift.org/ocp/<release> -o json 

Let me know what you think. Thanks very much!

Reply:
Oh yes, we should NEVER specify the release.
But we can tell the user how to extract the release:
./openshift-install version
will show the release which is used by the installer, for example:
```bash
[root@vm-9-193 ocp-virt-cluster]# ./openshift-install version
./openshift-install 4.9.0-0.nightly-2021-08-31-123131
built from commit 1d21be17667df4266c3def1de353488009881905
release image registry.ci.openshift.org/ocp/release@sha256:2cba6a2024037034a65f9bf3764a45a12c9e935b04385d409d437121508aa87b
release architecture amd64
```

So the user can use the release image from the openshift version command to understand which release to use.

The full flow would be:
```bash
[root@vm-9-193 ocp-virt-cluster]# ./openshift-install version|grep 'release image'

release image registry.ci.openshift.org/ocp/release@sha256:9aa0dce0b119d1d3f284df2cd30394a23e2d97f4b3e3853503a655caf90219a5
[root@vm-9-193 ocp-virt-cluster]# oc adm release info --registry-config pull-secret.txt registry.ci.openshift.org/ocp/release@sha256:9aa0dce0b119d1d3f284df2cd30394a23e2d97f4b3e3853503a655caf90219a5 -o json | jq -r '.references.spec.tags[] | select(.name == "baremetal-runtimecfg") | .from.name'

quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:b717191f780ebc5a9cce440993f51e0a2b6ef3eab3c450185996b4c79189506b
[root@vm-9-193 ocp-virt-cluster]# podman run --authfile /home/dev-scripts/pull_secret.json quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:b717191f780ebc5a9cce440993f51e0a2b6ef3eab3c450185996b4c79189506b
```

But notice that because we need the installation program to do that, then the section needs to be after the "Obtaining the installation program" [Link](https://deploy-preview-35316--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations.html#installation-obtaining-installer_installing-rhv-customizations)

Reply:
@dcdacosta what's `TAG` supposed to be? Please describe how to get the TAG.

Reply:
From the initial baremetal example [1] TAG is the release you are looking to install. 

[1] https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md

Reply:
Done.